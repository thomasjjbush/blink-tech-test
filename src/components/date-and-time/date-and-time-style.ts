import styled from 'styled-components';

export const Date = styled.span`
    color: inherit;
    font-size: 12px;
    font-weight: bold;
    margin: 0;
`;
