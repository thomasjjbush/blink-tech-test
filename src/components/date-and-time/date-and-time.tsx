import React, { FC, ReactElement } from 'react';
import { formatDate } from '../../utils/format-date/format-date';
import * as Styled from './date-and-time-style';

export const DateAndTime: FC = ({ children }: { children: string }): ReactElement => {
    return <Styled.Date>{formatDate(children)}</Styled.Date>;
};
