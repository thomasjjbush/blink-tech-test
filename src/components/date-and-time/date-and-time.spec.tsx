import React from 'react';
import { shallow } from 'enzyme';
import { DateAndTime } from './date-and-time';

describe('DateAndTime', () => {
    it('should match snapshot', () => {
        expect(shallow(<DateAndTime>{new Date(null).toString()}</DateAndTime>)).toMatchSnapshot();
    });

    it('should be null render if children resolves to false', () => {
        expect(shallow(<DateAndTime>{null}</DateAndTime>).isEmptyRender()).toBe(true);
    });
});
