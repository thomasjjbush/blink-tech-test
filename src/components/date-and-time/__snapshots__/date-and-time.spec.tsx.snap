// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateAndTime should match snapshot 1`] = `
<styled.span
  styles="
    color: inherit;
    font-size: 12px;
    font-weight: bold;
    margin: 0;
"
>
  1/1/1970 01:00
</styled.span>
`;
